src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java

```diff
@@ -255,21 +251,24 @@ public String getFieldSetFunctionName(TypeRef type, Name identifier) {
     }
   }
 
-  public String getResourceNameFieldSetFunctionName(TypeRef type, Name identifier) {
+  public String getResourceNameFieldSetFunctionName(FieldConfig fieldConfig, TypeRef type) {
```

Comment:
You could get by without the type parameter, and instead use `fieldConfig.getField().getType()`. (It looks like this pairing occurs several times in `SurfaceNamer`)